File: python-server/models/testimonial.py
from extensions import db, Base
from datetime import datetime

class Testimonial(Base):
    __tablename__ = 'testimonials'

    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
    user_id = db.Column(db.Integer, db.ForeignKey('user.id', ondelete='CASCADE'), nullable=False)
    quote = db.Column(db.Text, nullable=False)
    rating = db.Column(db.Integer, nullable=False)  # Rating from 1-5
    is_approved = db.Column(db.Boolean, default=False)  # Admin approval status
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    
    # User relationship
    user = db.relationship('User', backref='testimonials')
    
    def to_dict(self):
        return {
            'id': self.id,
            'user_id': self.user_id,
            'author': f"{self.user.first_name} {self.user.last_name}" if self.user.first_name and self.user.last_name else self.user.email,
            'quote': self.quote,
            'rating': self.rating,
            'is_approved': self.is_approved,
            'created_at': self.created_at.isoformat(),
            'updated_at': self.updated_at.isoformat()
        }

File: go-server/models/testimonial.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Testimonial struct {
	ID         uint           `json:"id" gorm:"primaryKey;autoIncrement"`
	UserID     uint           `json:"user_id" gorm:"not null;index;constraint:OnDelete:CASCADE" validate:"required"`
	Quote      string         `json:"quote" gorm:"not null;type:text" validate:"required"`
	Rating     int            `json:"rating" gorm:"not null" validate:"required,min=1,max=5"`
	IsApproved bool           `json:"is_approved" gorm:"default:false"`
	CreatedAt  time.Time      `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt  time.Time      `json:"updated_at" gorm:"autoUpdateTime"`
	DeletedAt  gorm.DeletedAt `json:"-" gorm:"index"`

	// Relationships
	User User `json:"user,omitempty" gorm:"foreignKey:UserID"`
}

// ToDict converts testimonial to map for JSON response
func (t *Testimonial) ToDict() map[string]interface{} {
	author := ""
	if t.User.FirstName != nil && t.User.LastName != nil {
		author = *t.User.FirstName + " " + *t.User.LastName
	} else {
		author = t.User.Email
	}

	return map[string]interface{}{
		"id":          t.ID,
		"user_id":     t.UserID,
		"author":      author,
		"quote":       t.Quote,
		"rating":      t.Rating,
		"is_approved": t.IsApproved,
		"created_at":  t.CreatedAt.Format(time.RFC3339),
		"updated_at":  t.UpdatedAt.Format(time.RFC3339),
	}
}

// TableName specifies the table name for GORM
func (Testimonial) TableName() string {
	return "testimonials"
}